.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: kdump-utils tests
+
+on: pull_request
+
+jobs:
+  format-check:
+    runs-on: ubuntu-latest
+    container: docker.io/fedora:latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: dnf install shfmt -y
+      - run: shfmt -d *.sh kdumpctl mk*dumprd
+
+  static-analysis:
+    runs-on: ubuntu-latest
+    container: docker.io/fedora:latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: dnf install shellcheck -y
+      # Currently, for kdump-utils, there is need for shellcheck to require
+      # the sourced file to give correct warnings about the checked file
+      - run: shellcheck --exclude=1090,1091 *.sh spec/*.sh kdumpctl mk*dumprd
```

Comment:
Instead of disabling SC1091 I would add option `-x` so shellcheck follows the external files. That prevents false positive errors, e.g. when a pull request defines a new variable in `kdump-lib.sh` and references it in `kdumpctl`. With #33 merged there should be a directive on all sourced files now, so shellcheck knows where to find the files. If there isn't I would consider it a bug in our code we should fix rather than simply hiding the error. Same for SC1090.

Plus as above we need to add the new directories.

Reply:
I disable SC109{0,1} in new version. New directories are also added now. Thanks!